lib/Event/NoteEvents.php

```diff
@@ -15,12 +16,12 @@
 
 namespace Pimcore\Event;
 
-final class ElementEvents
+final class NoteEvents
 {
     /**
-     * @Event("Pimcore\Event\Model\ElementEvent")
+     * @Event("Pimcore\Event\Model\NoteEvents")^
      *
      * @var string
      */
-    const POST_ADD = 'pimcore.element.note.postAdd';
+    const POST_ADD = 'pimcore.model.note.postAdd';
```

Comment:
The event gets triggered in https://github.com/pimcore/pimcore/blob/11.x/models/Element/Note.php#L166 - I changed the name since this is located in the models folder. And that is also the reason why we have a new "ModelEvent" here. The `ElementEvent` takes an `ElementInterface` parameter in the constructor. We pass `$this` to the constructor of the event, which leads to an error, since it is not derived from `ElementInterface`, but from `ModelInterface`.

Reply:
Change name to "pimcore.note.postAdd" like the naming we have in place for tags. Please see https://github.com/pimcore/pimcore/issues/13461 too.